src/lib/mongoBackend/mongoRegisterContext.cpp

```diff
@@ -67,8 +80,9 @@ HttpStatusCode mongoRegisterContext(RegisterContextRequest* requestP, RegisterCo
         id = OID(requestP->registrationId.get());
 
         mongoSemTake(__FUNCTION__, "findOne from RegistrationsCollection");
-        reg = connection->findOne(getRegistrationsCollectionName(tenant).c_str(), BSON("_id" << id));
+        reg = connection->findOne(getRegistrationsCollectionName(tenant).c_str(), BSON("_id" << id << "servicePath" << sPath));
```

Comment:
The function servicePathCheck in lib/common/ does this check (not accepting '#').
Called by postRegisterContext.cpp and postNotifyContextAvailability.cpp.
So, this should be covered.
However, I will add a func test with bad service path (I have one already, but more would be good).